backend/src/routes/observabilityProxy.ts

```diff
@@ -0,0 +1,98 @@
+/* Copyright Contributors to the Open Cluster Management project */
+import { constants, Http2ServerRequest, Http2ServerResponse, OutgoingHttpHeaders } from 'http2'
+import { request, RequestOptions } from 'https'
+import { pipeline } from 'stream'
+import { URL } from 'url'
+import { jsonRequest } from '../lib/json-request'
+import { logger } from '../lib/logger'
+import { notFound, unauthorized } from '../lib/respond'
+import { getCACertificate } from '../lib/serviceAccountToken'
+import { getToken } from '../lib/token'
+import { IResource } from '../resources/resource'
+
+const proxyHeaders = [
+  constants.HTTP2_HEADER_ACCEPT,
+  constants.HTTP2_HEADER_ACCEPT_ENCODING,
+  constants.HTTP2_HEADER_CONTENT_ENCODING,
+  constants.HTTP2_HEADER_CONTENT_LENGTH,
+  constants.HTTP2_HEADER_CONTENT_TYPE,
+]
+const proxyResponseHeaders = [
+  constants.HTTP2_HEADER_CACHE_CONTROL,
+  constants.HTTP2_HEADER_CONTENT_TYPE,
+  constants.HTTP2_HEADER_CONTENT_LENGTH,
+  constants.HTTP2_HEADER_CONTENT_ENCODING,
+  constants.HTTP2_HEADER_ETAG,
+]
+
+interface Route extends IResource {
+  spec: {
+    host?: string
+    path?: string
+    to?: {
+      kind?: 'Service'
+      name?: string
+      weight?: number
+    }
+    port?: {
+      targetPort?: string
+    }
+    tls?: {
+      termination?: 'edge' | 'passthrough' | 'reencrypt'
+      insecureEdgeTerminationPolicy?: 'Allow' | 'Disable' | 'Redirect'
+    }
+    wildcardPolicy?: 'Subdomain' | 'None'
+  }
+}
+
+export async function observabilityProxy(req: Http2ServerRequest, res: Http2ServerResponse): Promise<void> {
+  const token = getToken(req)
+  if (!token) return unauthorized(req, res)
+
+  const url = req.url
+
+  const headers: OutgoingHttpHeaders = { authorization: `Bearer ${token}` }
+  for (const header of proxyHeaders) {
+    if (req.headers[header]) headers[header] = req.headers[header]
+  }
+
+  const observabilityProxyRoute = await jsonRequest(
+    process.env.CLUSTER_API_URL +
+      `/apis/route.openshift.io/v1/namespaces/open-cluster-management-observability/routes/rbac-query-proxy`,
```

Comment:
Could the customer use a different namespace instead of `open-cluster-management-observability`? The documentation refers specifically to that namespace repeatedly, but I did not find any text that specifically says that namespace MUST be used. Is there an alternate way to find this route?

https://access.redhat.com/documentation/en-us/red_hat_advanced_cluster_management_for_kubernetes/2.8/html/observability/observing-environments-intro#enabling-observability-service

Reply:
I was going based on the documentation you posted above. I guess they could use another namespace? The only way I could think of finding this Route otherwise would be to query for every Route and then filter them based on ownerReferences (if the owner is `kind: MultiClusterObservability & name: observability`)

Reply:
FYI an update before the weekend.

I have tried to install Observability with a different namespace and am finding the Observability CRD never reaches healthy status as it cannot find the object storage secret (assuming due to it not being in the `open-cluster-management-obsevability` ns). 
Also, going through the obs repo, they have `open-cluster-management-observability` hardcoded all throughout the repo (see here: https://github.com/search?q=repo%3Astolostron%2Fmulticluster-observability-operator+open-cluster-management-observability&type=code).

I will ask on Monday if the namespace must be `open-cluster-management-observability` but as far as I can see it is.